src/main/java/org/traccar/protocol/Gl200TextProtocolDecoder.java

```diff
@@ -163,10 +204,17 @@ private Position decodeAck(Channel channel, SocketAddress remoteAddress, String[
 
     private Object decodeInf(Channel channel, SocketAddress remoteAddress, String sentence) {
         Parser parser = new Parser(PATTERN_INF, sentence);
-        Position position = initPosition(parser, channel, remoteAddress);
```

Comment:
We need to update this common method instead of doing a one off thing for INF type.

Reply:
ok, should we put the model in the position?

Reply:
No. There must be a better solution.

Reply:
ok, I think I need to remove parser.matches() from it right?

Reply:
can we put it in the deviceSession?

Reply:
I would avoid touching anything outside of this decoder, unless you want to go and update all other decoders.

Reply:
it's a little tricky, the method initialises the parser, gets the imei (which comes after the model) and returns a position. I don't see how we can refactor this.